### PR TITLE
FEATURE: Allow admins to force refresh "What's new?"

### PR DESCRIPTION
**Pull request by martin-brennan**
Sometimes changes to "What's new?" feed items are made or the feed items are
removed altogether, and the polling interval to check for new features is 1 day.

This is quite long, so this commit introduces a "Check for updates"
button for admins to click on the "What's new?" page which will bust
the cache for the feed and check again at the new features endpoint.
This is limited to 5 times per minute to avoid rapid sending of
requests.

image

### CODE REVIEW

**Comment by martin-brennan** [comment]
Chat system failure unrelated

File: app/assets/javascripts/admin/addon/controllers/admin-whats-new.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  checkForUpdates() {
+    if (this.checkFeaturesCallback) {
+      this.checkFeaturesCallback({ forceRefresh: true });
+    }
+  }
+
+  @action
+  bindCheckFeatures(checkFeaturesCallback) {
+    this.checkFeaturesCallback = checkFeaturesCallback;
+  }
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/controllers/admin-whats-new.js:15]: I'm not sold on this "pass an action between sibling components via the parent controller" pattern 😅

Thinking of alternatives: a dedicated service is too verbose imo; maybe an app-event?
(@davidtaylorhq, thoughts?)

File: app/assets/javascripts/admin/addon/components/dashboard-new-features.gjs [hunk 2]
```diff
@@ -49,24 +61,26 @@ export default class DashboardNewFeatures extends Component {
...
+            </AdminConfigAreaCard>
+          {{/each}}
+        {{else}}
+          <AdminConfigAreaEmptyList
+            @emptyLabelArgs={{hash
+              url="https://meta.discourse.org/tags/c/announcements/67/release-notes"
+            }}
+            @emptyLabel="admin.dashboard.new_features.previous_announcements"
+          />
+        {{/if}}
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/dashboard-new-features.gjs:82]: Oh nice!

File: app/assets/javascripts/admin/addon/components/dashboard-new-features.gjs [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
+
+  constructor() {
+    super(...arguments);
+    this.args.onCheckForFeatures(this.loadNewFeatures);
+  }
 
   @bind
-  loadNewFeatures() {
-    ajax("/admin/whats-new.json")
+  loadNewFeatures(opts = {}) {
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/dashboard-new-features.gjs:27]: Sure 👍

File: app/assets/javascripts/admin/addon/controllers/admin-whats-new.js (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/controllers/admin-whats-new.js:15]: Yeah I'm not sure I like it much either this just felt like the easiest option, happy to hear other suggestions.

File: app/assets/javascripts/admin/addon/components/admin-config-area-empty-list.gjs [hunk 4]
```diff
@@ -1,20 +1,33 @@
...
+      return i18n(this.args.emptyLabel, this.args.emptyLabelArgs);
+    }
 
-export default AdminConfigAreaEmptyList;
+    return i18n(this.args.emptyLabel);
+  }
+
+  <template>
+    <div class="admin-config-area-empty-list">
+      {{htmlSafe this.emptyLabel}}
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/admin/addon/components/admin-config-area-empty-list.gjs:18]: OK fair will change

File: config/locales/client.en.yml [hunk 5]
```diff
@@ -2861,6 +2861,7 @@ en:
       reaction_2: "<span>%{username}, %{username2}</span> %{description}"
       votes_released: "%{description} - completed"
       new_features: "New features available!"
+      check_for_updates: "Check for updates"
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply config/locales/client.en.yml:2864]: It's not, deleting

**Review by CvX — APPROVED** [review]

**Pull request merged**